src/editor/action-executor.ts

```diff
@@ -36,6 +39,67 @@ export class ActionExecutor {
         let preventDefaultEvent = true;
 
         switch (action.type) {
+            case EditActionType.InsertElseStatement: {
+                const newStatement = new ElseStatement(action.data.hasCondition);
+
+                if (action.data.outside) {
+                    // when the else is being inserted outside
+                    const elseRoot = context.lineStatement.rootNode as Module | Statement;
+                    newStatement.rootNode = elseRoot;
+                    newStatement.indexInRoot = context.lineStatement.indexInRoot;
+                    newStatement.body.push(new EmptyLineStmt(newStatement, 0));
+
+                    replaceInBody(elseRoot, newStatement.indexInRoot, newStatement);
+                    rebuildBody(elseRoot, newStatement.indexInRoot, context.lineStatement.lineNumber);
+                    this.module.editor.executeEdits(this.getBoundaries(context.lineStatement), newStatement);
+                } else {
+                    // when being inserted inside
+                    const curStmtRoot = context.lineStatement.rootNode as Statement;
+                    const elseRoot = curStmtRoot.rootNode as Module | Statement;
+                    newStatement.rootNode = elseRoot;
+                    newStatement.indexInRoot = curStmtRoot.indexInRoot + 1;
+
+                    // indent back and place all of the code below it as its child
+                    const toMoveStatements = curStmtRoot.body.splice(
```

Comment:
Scope references need to be updated here or after this happens. Since `IfStatement` and `ElseStatement` have separate scopes now. Just leaving this as a note so we don't forget. 

What needs to happen exactly is that all references that were part of the scope of the statement above need to be taken out of that scope and placed inside the scope of the newly created Elif or Else statement.

Reply:
That's true. let me see if I can fix this issue along with changing indentations. Should be straightforward.